autotest/cpp/test_gdal_jpegxr.cpp

```diff
@@ -0,0 +1,389 @@
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+            ensure_equals(GDALGetRasterDataType(GDALGetRasterBand(ds_.h, i + 1)), GDT_Byte);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 1)), GCI_RedBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 2)), GCI_GreenBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 3)), GCI_BlueBand);
+    }
+
+    // OPEN: 32bpp BGRA
+    template<>
+    template<>
+    void object::test<5>()
+    {
+        std::string file = get_file_path("lenna-256x256-32bpp-BGRA.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+        ensure_equals(GDALGetRasterXSize(ds_.h), 256);
+        ensure_equals(GDALGetRasterYSize(ds_.h), 256);
+        ensure_equals(GDALGetRasterCount(ds_.h), 4);
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+            ensure_equals(GDALGetRasterDataType(GDALGetRasterBand(ds_.h, i + 1)), GDT_Byte);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 1)), GCI_BlueBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 2)), GCI_GreenBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 3)), GCI_RedBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 4)), GCI_AlphaBand);
+    }
+
+    // OPEN: 32bpp RGBA
+    template<>
+    template<>
+    void object::test<6>()
+    {
+        std::string file = get_file_path("lenna-256x256-32bpp-RGBA.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+        ensure_equals(GDALGetRasterXSize(ds_.h), 256);
+        ensure_equals(GDALGetRasterYSize(ds_.h), 256);
+        ensure_equals(GDALGetRasterCount(ds_.h), 4);
+        for (int i = 0; i < GDALGetRasterCount(ds_.h); i++)
+            ensure_equals(GDALGetRasterDataType(GDALGetRasterBand(ds_.h, i + 1)), GDT_Byte);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 1)), GCI_RedBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 2)), GCI_GreenBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 3)), GCI_BlueBand);
+        ensure_equals(GDALGetRasterColorInterpretation(GDALGetRasterBand(ds_.h, 4)), GCI_AlphaBand);
+    }
+
+    // READ ////////////////////////////////////////////////////////////////////
+
+    // READ: 8bpp Gray
+    template<>
+    template<>
+    void object::test<7>()
+    {
+        std::string file = get_file_path("lenna-256x256-8bpp-Gray.jxr");
+        ds_ = GDALOpen(file.c_str(), GA_ReadOnly);
+        ensure(ds_.h);
+
+        double dfMin(0), dfMax(0), dfMean(0), dfStdD(0);
+        GDALRasterBandH hBand1 = GDALGetRasterBand(ds_.h, 1);
+        CPLErr eErr = GDALGetRasterStatistics(hBand1, false, true, &dfMin, &dfMax, &dfMean, &dfStdD);
```

Comment:
I'd rather use GDALChecksumImage() and checks the value of the checksum to ensure proper decoding is done. Same for similar tests